src/components/learner-credit-management/data/utils.js

```diff
@@ -551,3 +555,98 @@ export const isLmsBudget = (
   activeIntegrationsLength,
   isUniversalGroup,
 ) => activeIntegrationsLength > 0 && isUniversalGroup;
+
+/**
+ * Determines if the course has already started. Mostly used around text formatting for tense
+ *
+ * This should also help with reducing 'flaky' tests.
+ *
+ * @param start
+ * @returns {boolean}
+ */
+export const hasCourseStarted = (start) => dayjs(start).isBefore(dayjs());
+
+/**
+ * Returns assignable course runs within the threshold of within the subsidies expiration date
+ * offset by the DAYS_UNTIL_ASSIGNMENT_ALLOCATION_EXPIRATION constant. It sorts it from the soonest expiring
+ * enroll-by date and the enroll-by date and upgrade deadline has been normalized to ISO format.
+ *
+ * @param courseRuns
+ * @param subsidyExpirationDatetime
+ * @param isLateRedemptionAllowed
+ * @returns {*}
+ */
+export const getAssignableCourseRuns = ({ courseRuns, subsidyExpirationDatetime, isLateRedemptionAllowed }) => {
+  const clonedCourseRuns = courseRuns.map(a => ({
+    ...a,
+    enrollBy: dayjs.unix(a.enrollBy).toISOString(),
+    upgradeDeadline: dayjs.unix(a.upgradeDeadline).toISOString(),
```

Comment:
[clarification] Is `upgradeDeadline` needed here if the related `assignableCourseRunsFilter` does not consume the `upgradeDeadline`?

Reply:
This change is part of a nice to have for future development where the current `upgradeDeadline` value is also a timestamp value.